axum-extra/src/response/mod.rs

```diff
@@ -9,3 +9,82 @@ pub use erased_json::ErasedJson;
 #[cfg(feature = "json-lines")]
 #[doc(no_inline)]
 pub use crate::json_lines::JsonLines;
+
+macro_rules! mime_response {
+    (
+        $(#[$m:meta])*
+        $ident:ident,
+        $mime:ident,
+    ) => {
+        mime_response! {
+            $(#[$m])*
+            $ident,
+            mime::$mime.as_ref(),
+        }
+    };
+
+    (
+        $(#[$m:meta])*
+        $ident:ident,
+        $mime:expr,
+    ) => {
+        $(#[$m])*
+        #[derive(Clone, Copy, Debug)]
+        #[must_use]
+        pub struct $ident<T>(pub T);
+
+        impl<T> axum::response::IntoResponse for $ident<T>
+        where
+            T: axum::response::IntoResponse,
+        {
+            fn into_response(self) -> axum::response::Response {
+                (
+                    [(
+                        http::header::CONTENT_TYPE,
+                        http::HeaderValue::from_static($mime),
```

Comment:
Maybe put in a `const` so it's instantiated at compile time?

Reply:
Can't do that since `$mime` isn't `const` because we it from [`mime.as_ref()`](https://docs.rs/mime/latest/mime/struct.Mime.html#impl-AsRef%3Cstr%3E-for-Mime).